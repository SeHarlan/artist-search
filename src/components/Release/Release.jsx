import React from 'react';
import { useFetchSongs } from '../../services/hooks';
import withList from '../HOCs/WithList/WithList';
import Song from '../Songs/Song';
import placeholder from '../../assets/placeholder.png';
import { useParams } from 'react-router-dom';
import style from './release.css';

const Release = () => {
  const { releaseTitle, releaseId, artist } = useParams();

  const img = `http://coverartarchive.org/release/${releaseId}/front`;
  const handleImgError = ({ target }) => (target.src = placeholder);
  
  const songsData = useFetchSongs(releaseId, artist);
  const Songs = withList(Song);

  return (
    <section className={style.Release}>
      <h2>{releaseTitle}</h2>
      <img src={img} onError={handleImgError} alt={releaseTitle}/>
      <Songs list={songsData}/>
    </section>
  );
};

export default Release;
