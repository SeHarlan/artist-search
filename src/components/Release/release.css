.Release {
  h2 {
    text-align: center;
    text-decoration: underline;
  }
  img {
    display: block;
    margin: 20px auto;
    width: 25vw;
    border: 4px outset wheat;
  }
  ul {
    margin: 0 30vw;
    padding: 1rem;
    list-style: none;
    text-align: center;
    border-radius: 1rem;
    box-shadow: inset 0 0 10px rgba(0, 0, 0, 0.3);
  }
  li {
    margin: 0.5rem;
  }
}