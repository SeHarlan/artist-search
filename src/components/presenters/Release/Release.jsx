import React from 'react';
import PropTypes from 'prop-types';
import { useFetchSongs } from '../../../services/hooks';
import withList from '../../HOCs/WithList/WithList';
import Song from '../Songs/Song';

const Release = ({ match }) => {
  const { releaseTitle, releaseId, imageBool, artist } = match.params;
  const title = releaseTitle;
  const isThereImage = imageBool === 'true';

  const img = isThereImage ? `http://coverartarchive.org/release/${releaseId}/front` : '/placeholder.png';
  
  const songsData = useFetchSongs(releaseId, artist);
  const Songs = withList(Song);

  return (
    <>
      <h2>{title}</h2>
      <img src={img} alt={title}/>
      <Songs list={songsData}/>
    </>
  );
};

Release.propTypes = {
  match: PropTypes.object.isRequired
};

export default Release;
