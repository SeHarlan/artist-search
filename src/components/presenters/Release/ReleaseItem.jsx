import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';


const ReleaseItem = ({ releaseTitle, releaseId, imageBool, artist }) => {

  const image = imageBool ? `http://coverartarchive.org/release/${releaseId}/front` : '/placeholder.png';
  return (
    <Link to={`/releases/${artist}/${releaseTitle}/${releaseId}/${imageBool}`} >
      <h2>{releaseTitle}</h2>
      <img src={image} alt={releaseTitle} />
    </Link>
  );
};


ReleaseItem.propTypes = { 
  releaseTitle: PropTypes.string.isRequired, 
  releaseId: PropTypes.string.isRequired, 
  imageBool: PropTypes.bool.isRequired, 
  artist:PropTypes.string.isRequired
};

export default ReleaseItem;
