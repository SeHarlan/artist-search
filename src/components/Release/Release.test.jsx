import React from 'react';
import { shallow } from 'enzyme';
import Release from './Release';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    releaseTitle: 'test', 
    releaseId: 'test', 
    artist: 'test'
  }),
  useRouteMatch: () => ({ url: '/releases/test/test/test' }),
}));

describe('Release component', () => {
  it('renders Release', () => {
    const wrapper = shallow(<Release />);
    expect(wrapper).toMatchSnapshot();
  });
});
