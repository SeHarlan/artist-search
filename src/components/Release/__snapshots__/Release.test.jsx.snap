// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release component renders Release 1`] = `
<section
  className="Release"
>
  <h2>
    test
  </h2>
  <img
    alt="test"
    onError={[Function]}
    src="http://coverartarchive.org/release/test/front"
  />
  <WithList(Song)
    list={Array []}
  />
</section>
`;
